File: typescript/src/types/metadata.ts
// Object-level metadata.
// Metadata for specific LiveObject types is nested under the appropriate key.
export interface ObjectMetadata {
  objectId: string;
  siteTimeserials: Record<string, string>;
  tombstone: boolean;
  map?: {
    semantics: "LWW";
  };
}

// Entry-level metadata on entries in collection types.
export interface EntryMetadata {
  timeserial: string;
  tombstone: boolean;
}

File: swift/Sources/AblyLiveObjects/metadata.swift
// Object-level metadata.
// Metadata for specific LiveObject types is nested under the appropriate key.
public protocol ObjectMetadata {
    var objectId: String { get }
    var siteTimeserials: [String: String] { get }
    var tombstone: Bool { get }
    // Lawrence: probably would have a separate type for this, not important now
    //  map?: {
//    semantics: "LWW";
    //  };
}

// Entry-level metadata on entries in collection types.
public protocol EntryMetadata {
    var timeserial: String { get }
    var tombstone: Bool { get }
}
